front/app/containers/ProjectsShowPage/timeline/index.tsx

```diff
@@ -111,50 +109,31 @@ const ProjectTimelineContainer = memo<Props & WithRouterProps>(
 
     useEffect(() => {
       if (
-        !isNilOrError(location) &&
+        selectedPhase !== null &&
         !isNilOrError(phases) &&
-        phases.length > 0
+        project &&
+        !isNilOrError(locale)
       ) {
-        const currentPhase = getCurrentPhase(phases);
-        const firstPhase = getFirstPhase(phases);
-        const lastPhase = getLastPhase(phases);
-        const lastPastPhase = getLastPastPhase(phases);
-
-        // if, coming from the siteMap, a phase url parameter was passed in, we pick that phase as the default phase,
-        // then remove the param so that when the user navigates to other phases there is no mismatch
-        if (isString(location?.query?.phase)) {
-          const phase = phases.find(
-            (phase) => phase.id === location.query.phase
-          );
-
-          if (phase) {
-            window.history.replaceState(null, '', location.pathname);
-            selectPhase(phase);
-          }
-        } else if (currentPhase) {
-          selectPhase(currentPhase);
-        } else if (
-          firstPhase &&
-          pastPresentOrFuture([
-            firstPhase.attributes.start_at,
-            firstPhase.attributes.end_at,
-          ]) === 'future'
-        ) {
-          selectPhase(firstPhase);
-        } else if (
-          lastPastPhase &&
-          lastPhase &&
-          pastPresentOrFuture([
-            lastPhase.attributes.start_at,
-            lastPhase.attributes.end_at,
-          ]) === 'future'
-        ) {
-          selectPhase(lastPastPhase);
+        setPhaseURL(selectedPhase.id, phases, project, locale);
+      }
+    }, [selectedPhase]);
+
+    useEffect(() => {
+      if (!isNilOrError(phases) && phases.length > 0) {
+        const latestRelevantPhase = getLatestRelevantPhase(phases);
+
+        // if a phase parameter was provided, and it is valid, we set that as phase.
+        // otherwise, use the most logical phase
+        if (isValidPhase(phaseNumber, phases)) {
+          const phaseIndex = Number(phaseNumber) - 1;
+          selectPhase(phases[phaseIndex]);
+        } else if (latestRelevantPhase) {
+          selectPhase(latestRelevantPhase);
         } else {
-          selectPhase(lastPhase || null);
+          selectPhase(null);
         }
       }
-    }, [location, phases]);
+    }, [phases]);
```

Comment:
Also a dependency on phaseNumber here.

Reply:
Same as above: `phaseNumber` is only relevant on page load, after that it won't change unless the whole page reloads (i.e. you enter a new URL)

Reply:
> @brentguf I am not sure, it depends on the situation I guess. Some users will want to link a specific phase regardless of whether it is the active phase, whereas other users will just want to always link to the active phase. If the latter type of users go back to an earlier phase just to check the result or whatever, and then back to the active phase, that specific phase will suddenly be locked into the URL, which might not be their intention.
> 
> I personally don't know which situation will be more common. @juliennechen ?

Yes, but then they have actively selected a phase, so can expect it to be in the URL. In any case, currently it's not possible to link to the active phase, which I would say is a problem. If they say, "vote now in our Vote Phase!" (which would include a link) and then afterwards when the phase is done someone clicks the link and are linked to the current active phase where they can't vote, it'll be confusing. You should be able to link to the current phase without manually having to edit the URL. 

Later we can add a button 'copy project URL' to have the link without phases if that helps.

@kogre since you made the initial comment, what do you think here?

Reply:
Or if you click the timeline (or navigation arrows), but that's not relevant for this useEffect. This one is only for the initial phase selection, navigation is handled by a different component (PhaseNavigation). So I think you can leave it out here indeed.